core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java

```diff
@@ -82,13 +108,39 @@ public NotificationCenter() {
     // private list of notification by notification type.
     // we used a list so that notification order can mean something.
     private Map<NotificationType, ArrayList<NotificationHolder>> notificationsListeners = new HashMap<NotificationType, ArrayList<NotificationHolder>>();
+    private List<NotificationHolder> decisionListenerHolder = new ArrayList<>();
+
+    /**
+     * Convenience method to support lambdas as callbacks in later version of Java (8+).
+     *
+     * @param decisionNotificationListener
+     * @return greater than zero if added.
+     */
+    public int addDecisionNotificationListener(DecisionNotificationListener decisionNotificationListener) {
+        if (decisionNotificationListener != null) {
+            for (NotificationHolder holder : decisionListenerHolder) {
+                if (holder.decisionNotificationListener == decisionNotificationListener) {
+                    // TODO: 3/27/2019 change log level from warn to info and to return existing listener ID
```

Comment:
Is this relevant still?

Reply:
yes we need to fix it in separate PR as Mike davis suggested